nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java

```diff
@@ -136,14 +136,19 @@ public void event(Event event) {
             if (null == threadPoolExecutor) {
                 initialiseThreadPools();
             }
+        } else if (event.is(EventTypes.PREPARE_SHUTDOWN)) {
+            stopThreadPools();
```

Comment:
I'm not sure if this is not too early -- prepare shutdown is broadcasted before applications are stopped, as initlevel is decreased after broadcasting this event.
Therefore applications might be failing during their stop.

Reply:
agree this is probably too early